torch/csrc/jit/type.h

```diff
@@ -215,35 +239,41 @@ struct TORCH_API ListType : public Type {
   static TypePtr ofTensors();
   static TypePtr ofInts();
 private:
+  ListType(TypePtr elem)
+  : Type(TypeKind::ListType), elem(elem) {}
+  static const TypeKind Kind = TypeKind::ListType;
```

Comment:
This can probably be constexpr

Reply:
Thanks! I will wrap the optimizations in another PR as the CI build took too long and don't want this to block other's work.